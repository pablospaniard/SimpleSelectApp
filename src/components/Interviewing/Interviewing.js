import React from 'react'
import {
  Table,
  TableBody,
  TableCell,
  TableHead,
  TableRow,
  Checkbox,
  Typography
} from '@material-ui/core'
import { withStyles } from '@material-ui/core/styles'
import PropTypes from 'prop-types'

const styles = theme => ({
  root: {
    width: '100%',
    marginTop: theme.spacing.unit * 3,
    overflowX: 'auto'
  },
  table: {
    minWidth: 700
  }
})

const Interviewing = props => {
  const { persons, classes, onChangeHandler, selected } = props
  if (selected.length === 0) {
    return (
      <Typography variant="subheading" style={{ padding: '20px' }}>
        You not selected any candidates...
      </Typography>
    )
  } else
    return (
      <Table className={classes.table}>
        <TableHead>
          <TableRow>
            <TableCell>Foto</TableCell>
            <TableCell>Full Name</TableCell>
            <TableCell>Email</TableCell>
            <TableCell>CellPhone</TableCell>
            <TableCell>Nationality</TableCell>
            <TableCell />
          </TableRow>
        </TableHead>
        <TableBody>
          {persons.map(person => {
            if (selected.indexOf(person.email) >= 0) {
              return (
                <TableRow key={person.id.value}>
                  <TableCell>
                    <img src={`${person.picture.thumbnail}`} alt="foto" />
                  </TableCell>
                  <TableCell component="th" scope="row">
                    {`${person.name.first} ${person.name.last}`}
                  </TableCell>
                  <TableCell>{person.email}</TableCell>
                  <TableCell>{person.cell}</TableCell>
                  <TableCell>{person.nat}</TableCell>
                  <TableCell>
                    <Checkbox onChange={() => onChangeHandler(person.email)} />
                  </TableCell>
                </TableRow>
              )
            } else return null
          })}
        </TableBody>
      </Table>
    )
}

Interviewing.propTypes = {
  persons: PropTypes.array.isRequired,
  onChangeHandler: PropTypes.func,
  classes: PropTypes.object,
  selected: PropTypes.array.isRequired
}

export default withStyles(styles)(Interviewing)
