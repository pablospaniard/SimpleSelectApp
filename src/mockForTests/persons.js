export const persons = [
  {
    gender: 'female',
    name: {
      title: 'ms',
      first: 'nicky',
      last: 'vargas'
    },
    location: {
      street: '9761 station road',
      city: 'salisbury',
      state: 'cumbria',
      postcode: 'NN4 9LF',
      coordinates: {
        latitude: '-5.3269',
        longitude: '48.1334'
      },
      timezone: {
        offset: '+2:00',
        description: 'Kaliningrad, South Africa'
      }
    },
    email: 'nicky.vargas@example.com',
    login: {
      uuid: 'bee8fcb3-1261-47ff-ab6f-75c7be01077c',
      username: 'yellowladybug417',
      password: 'gogogo',
      salt: 'ixgD083Q',
      md5: '3c6400fffd6adddba4df6856079d61fd',
      sha1: '8f109908f4db48a9a8743ecf1ecc0d66e2536888',
      sha256: 'be9fd8363e25c08f58fb111600673a3c548b9f158fc847b5c48212f5fb2fa3c3'
    },
    dob: {
      date: '1986-01-13T11:28:23Z',
      age: 32
    },
    registered: {
      date: '2009-09-11T00:05:01Z',
      age: 9
    },
    phone: '016973 35595',
    cell: '0707-619-299',
    id: {
      name: 'NINO',
      value: 'TZ 01 18 38 X'
    },
    picture: {
      large: 'https://randomuser.me/api/portraits/women/69.jpg',
      medium: 'https://randomuser.me/api/portraits/med/women/69.jpg',
      thumbnail: 'https://randomuser.me/api/portraits/thumb/women/69.jpg'
    },
    nat: 'GB'
  }
]
