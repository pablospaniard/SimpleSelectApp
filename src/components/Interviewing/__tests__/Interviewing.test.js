import React from 'react'
import { render, cleanup } from 'react-testing-library'
import Interviewing from '../Interviewing'
import { persons } from '../../../mockForTests/persons'

describe('Interviewing should render properly', () => {
  afterEach(cleanup)

  const selected = []

  it('should match snapshot', () => {
    const { container } = render(
      <Interviewing persons={persons} selected={selected} />
    )
    expect(container.firstChild).toMatchSnapshot('Interviewing_snapshot_1')
  })
})
