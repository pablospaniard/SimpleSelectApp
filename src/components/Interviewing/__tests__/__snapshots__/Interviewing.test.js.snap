// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Interviewing_snapshot_1 1`] = `
<h3
  class="MuiTypography-root-3 MuiTypography-subheading-10"
  style="padding: 20px;"
>
  You not selected any candidates...
</h3>
`;
